src/wallet/test/coinselector_tests.cpp

```diff
@@ -724,12 +724,25 @@ BOOST_AUTO_TEST_CASE(waste_test)
     BOOST_CHECK_LT(waste_nochange2, waste_nochange1);
     selection.clear();
 
-    // 0 Waste only when fee == long term fee, no change, and no excess
+    // 0 Waste when fee == long term fee, no change, and no excess
     add_coin(1 * COIN, 1, selection, fee, fee);
     add_coin(2 * COIN, 2, selection, fee, fee);
     const CAmount exact_target = in_amt - 2 * fee;
     BOOST_CHECK_EQUAL(0, GetSelectionWaste(selection, 0, exact_target));
+    selection.clear();
 
+    // 0 Waste when (fee - long term fee) == cost of change, and no excess
```

Comment:
```suggestion
    // 0 Waste when (long term fee - fee) == cost of change, and no excess
```
As it is currently written the two aren't equal but negatives of each other.